src/Page.js

```diff
@@ -227,24 +242,76 @@ function getClosestHeading($, headingsSelector, element) {
 }
 
 /**
- * Records headings and keywords inside rendered page into this.headings and this.keywords respectively
+ * Checks if page.frontMatter has a valid page navigation specifier
+ * @returns boolean
  */
-Page.prototype.collectHeadingsAndKeywords = function () {
-  const $ = cheerio.load(fs.readFileSync(this.resultPath));
-  this.collectHeadingsAndKeywordsInContent($(`#${CONTENT_WRAPPER_ID}`).html(), null, false);
+Page.prototype.validPageNavigationSpecifier = function () {
+  const { pageNav } = this.frontMatter;
+  return pageNav && (pageNav === 'default' || Number.isInteger(pageNav));
 };
 
 /**
- * Generates a heading selector based on the indexing level
- * @param headingIndexingLevel to generate
+ * Collect headings outside of models and panels.
+ * @param content, html content of a page
  */
-function generateHeadingSelector(headingIndexingLevel) {
-  let headingsSelector = 'h1';
-  for (let i = 2; i <= headingIndexingLevel; i += 1) {
-    headingsSelector += `, h${i}`;
+Page.prototype.collectNavigableHeadings = function (content) {
+  if (this.validPageNavigationSpecifier()) {
+    const { pageNav } = this.frontMatter;
+    let elementSelector;
+    // Use specified navigation level or default in this.headingIndexingLevel
+    if (pageNav === 'default') {
+      elementSelector = generateHeadingSelector(this.headingIndexingLevel);
+    } else if (Number.isInteger(pageNav)) {
+      elementSelector = generateHeadingSelector(parseInt(pageNav, 10));
+    } else {
+      // Not a valid specifier
+      return;
+    }
```

Comment:
How about like this, we refactor out this logic:

```js
Page.prototype.generateElementSelectorForPageNav = function (pageNav) {
  if (pageNav === 'default') {
    return generateHeadingSelector(this.headingIndexingLevel);
  } else if (Number.isInteger(pageNav)) {
    return = generateHeadingSelector(parseInt(pageNav, 10));
  } else {
    // Not a valid specifier
    return undefined;
  }
}
```

Then:

* `collectNavigableHeadings()` can directly call this method. `isPageNavigationSpecifierValid()` will just return 
* `generateElementSelectorForPageNav(pageNav) !== undefined`.

Justification: To avoid duplicating the logic of handling pageNav as much as possible (lest we forget to change both places if there's a change in the behaviour of pageNav in the future).

Reply:
This implementation will result in unnecessary processing since `generateElementSelectorForPageNav(pageNav)` will invoke `generateHeadingSelector` if the specifier is valid.

We can still do the `generateElementSelectorForPageNav(pageNav)` abstraction to avoid duplicate logic within `collectNavigableHeadings`. 

```Javascript
  const { pageNav } = this.frontMatter;
  const elementSelector = this.generateElementSelectorForPageNav(pageNav);
  if (elementSelector === undefined) {
    return;
  }
  // Rest of function
```
To make the specifier checking more efficient, maybe we can store the result of `isPageNavigationSpecifierValid()` in a variable (`this.hasValidPageNavigationSpecifier`)?

Reply:
> This implementation will result in unnecessary processing since `generateElementSelectorForPageNav(pageNav)` will invoke `generateHeadingSelector` if the specifier is valid.

You are right, it is indeed inefficient.

> To make the specifier checking more efficient, maybe we can store the result of `isPageNavigationSpecifierValid()` in a variable (`this.hasValidPageNavigationSpecifier`)?

Not sure whether it is worth going this far actually.

I guess the original code is OK then, sometimes it necessary to have the "duplicates" in order to get the trade offs.